docs/ocis/adr/0021-service-authorization.md

```diff
@@ -0,0 +1,127 @@
+
+#### Where do we store them? In the same user backend?
+In the CS3 API we can find USER_TYPE_SERVICE and USER_TYPE_APPLICATION, where application is something like collabora and service is afaict sth like service accounts.
+
+#### Do we have to implement a new reva auth-service manager?
+
+#### Do graph permission names have to bleed into the internal CS3 implementation?
+Maybe not, AFAICT we could internally use spicedb and use it to implement the permission checks.
+
+#### Why can't we just mint a token and bake the permissions the service has into it.
+For normal users this is called the scope and it covers the permissions the use has consented to. When an access token arrives at the proxy we can use the scope claim which is filled by the IdP to limit the request to that scope. However, we will not be able to teach every IdP every scope that users should be able to consent to. It is far more practical to read a role claim and mint the scopes based on that, e.g. a role user would get the `Files.ReadWrite.All` permission that scopes the request to read all his files. A service that uses delegated permission can during this request only access the users files and all files the user has access to. We can add a scope middleware to the graph api (or other http endpoints) that checks if the user has consonted to the scope required for the request.
+
+This does not free us from the actual permission check to determine if the user can actually read or write the file. While `Files.ReadWrite.All` means that the service could write to a file that was shared with the logged in user an actual permission check needs to determine if the file was shared with write permission or if it was read only.
+
+#### How do we check the service account has the permission to make a call?
+In the same way we currently check permissions. We delegate that to the permissions service. Hm but that means we need to make two permission checks: one for the service account and one for the logged in user, if present. We need to determine if the service is allowed to make a request and if the user is allowed to make a request. We want to protect against .g. a compromised thumbnail service that should only make read requests but then tries to write a file. This needs to be checked at the storage provider.
+
+What if we logically AND the delegated permission with the users scope?
+
+How does go micro solve this?
+
+To be clear: this has nothing to do with the scope.
+
+
+#### How do we "provision" service accounts?
+
+#### How/where do we assign roles/permissions to them?
```

Comment:
replace "to them" by "to service accounts" probably.

Reply:
Moved description to the Service Accounts section.